modyn/selector/internal/selector_strategies/downsampling_strategies/rho_loss_downsampling_strategy.py

```diff
@@ -51,34 +51,43 @@ def inform_next_trigger(self, next_trigger_id: int, selector_storage_backend: Ab
     def downsampling_params(self) -> dict:
         config = super().downsampling_params
         config["rho_pipeline_id"] = self.rho_pipeline_id
+        il_model_id = self._get_latest_il_model_id(self.rho_pipeline_id, self._modyn_config)
+        assert il_model_id is not None
+        config["il_model_id"] = il_model_id
+        return config
 
-        with MetadataDatabaseConnection(self._modyn_config) as database:
+    @staticmethod
+    def _get_latest_il_model_id(rho_pipeline_id: int, modyn_config: dict) -> Optional[int]:
+        with MetadataDatabaseConnection(modyn_config) as database:
             # find the maximal trigger id. This is the current trigger id.
             max_trigger_id = (
                 database.session.query(func.max(Trigger.trigger_id))
-                .filter(Trigger.pipeline_id == self.rho_pipeline_id)
+                .filter(Trigger.pipeline_id == rho_pipeline_id)
                 .scalar()
             )
-            assert max_trigger_id is not None
+            if max_trigger_id is None:
+                return None
 
             # one pipeline id and one trigger id can only correspond to one model
             il_model_id = (
                 database.session.query(TrainedModel.model_id)
-                .filter(TrainedModel.pipeline_id == self.rho_pipeline_id, TrainedModel.trigger_id == max_trigger_id)
+                .filter(TrainedModel.pipeline_id == rho_pipeline_id, TrainedModel.trigger_id == max_trigger_id)
                 .scalar()
             )
             assert il_model_id is not None
-
-        config["il_model_id"] = il_model_id
-        return config
+        return il_model_id
 
     def _train_il_model(self, trigger_id: int) -> int:
+        if self.il_training_config.use_previous_model:
```

Comment:
Nit but we can put this into a single statement like a = b if c else d

Reply:
yeah that's my first choice but I feel I will have a really long single line so I switched to this old-school style :D